### PR TITLE
chore: adapt queryclient to multi cloud

### PR DESCRIPTION
**Pull request by jwcesign**
Fixes #none

## Proposed Changes

* adapt queryclient to multi cloud

**Release Note**

```release-note

```

### CODE REVIEW

**Comment by jwcesign** [comment]
cc @helen-frank

File: pkg/tools/queryclient.go [hunk 1]
```diff
@@ -22,21 +23,31 @@ type QueryClientInterface interface {
...
 
-	awsMutex     sync.Mutex
-	awsPriceData map[string]*apis.RegionalInstancePrice
+	awsMutex  sync.Mutex
+	priceData map[string]*apis.RegionalInstancePrice
 }
 
-func NewQueryClient(awsEndpoint, region string) (QueryClientInterface, error) {
+func NewQueryClient(endpoint, cloudProvider, region string) (QueryClientInterface, error) {
+	queryBaseUrl, err := url.JoinPath(endpoint, "/api/v1/aws/ec2")
```
Comment (helen-frank) [inline pkg/tools/queryclient.go:36]: use switch

Comment (jwcesign) [reply pkg/tools/queryclient.go:42]: fixed, please check

**Pull request merged**